### PR TITLE
Products comments

### PR DESCRIPTION
**Pull request by emord**
comments from https://github.com/dimagi/commcare-hq/pull/6805

@czue @esoergel

### CODE REVIEW

**Comment by czue** [comment]
nice! thanks for followup

File: corehq/apps/products/fixtures.py [hunk 1]
```diff
@@ -18,16 +18,16 @@
...
 
     fields = filter(lambda x: x != CUSTOM_DATA_SLUG, PRODUCT_FIELDS)
     fields.append('@id')
 
-    custom_fields = CustomDataFieldsDefinition.get_by_domain_and_type(domain, 'ProductFields')
+    custom_fields = get_by_domain_and_type(domain, 'ProductFields')
     if custom_fields:
-        for f in custom_fields.fields:
-            fields.append(CUSTOM_DATA_SLUG + '/' + f.slug)
+        for f in custom_fields['doc']['fields']:
```
Comment (czue) [inline corehq/apps/products/fixtures.py]: one subtle change with this is that if the keys don't exist on the doc then this will fail hard (whereas jsonobject will nicely deal with that). will let @esoergel merge as long as we're never worried this data being messy.

File: corehq/apps/custom_data_fields/dbaccessors.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        'custom_data_fields/by_field_type',
+        key=[domain, field_type],
+        include_docs=True,
+        reduce=False,
+        # if there's more than one,
+        # it's probably because a few were created at the same time
+        # due to a race condition
+        # todo: a better solution might be to use locking in this code
+        limit=1,
+    ).one()
```
Comment (czue) [inline corehq/apps/custom_data_fields/dbaccessors.py]: might be easier for callers of this function if you return ['doc'] though there may be some edge cases to test for if it doesn't return anything

File: corehq/apps/products/fixtures.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/products/fixtures.py]: What's the advantage of passing around the unwrapped doc?  It seems like it'd just be simpler to let jsonobject worry about it (although I don't think this data will be messy, it's schema'd and hasn't changed that I'm aware of).

File: corehq/apps/custom_data_fields/dbaccessors.py (hunk 2 above)
Comment (esoergel) [inline corehq/apps/custom_data_fields/dbaccessors.py]: Just commented on this below, but curious why this is doing `get_db().view` instead of returning a wrapped doc.

Comment (emord) [inline corehq/apps/custom_data_fields/dbaccessors.py]: Was originally thinking it'd be faster to not wrap it. Silly premature optimization. Returning a wrapped doc now
